plugins/pastetools/filter/image.js

```diff
@@ -300,13 +390,50 @@
 	}
 
 	function createSrcWithBase64( img ) {
-		var isSupportedType = CKEDITOR.tools.array.indexOf( CKEDITOR.pasteFilters.image.supportedImageTypes, img.type ) !== -1;
+		var isSupportedType = CKEDITOR.tools.array.indexOf( CKEDITOR.pasteFilters.image.supportedImageTypes, img.type ) !== -1,
+			data = img.hex;
 
 		if ( !isSupportedType ) {
 			return null;
 		}
 
-		return img.type ? 'data:' + img.type + ';base64,' + CKEDITOR.tools.convertBytesToBase64( CKEDITOR.tools.convertHexStringToBytes( img.hex ) ) : null;
+		if ( typeof data === 'string' )  {
+			data = CKEDITOR.tools.convertHexStringToBytes( img.hex );
+		}
+
+		return img.type ? 'data:' + img.type + ';base64,' + CKEDITOR.tools.convertBytesToBase64( data ) : null;
+	}
+
+	function convertBlobUrlToBase64( blobUrlSrc ) {
+		return new CKEDITOR.tools.promise( function( resolve ) {
+			CKEDITOR.ajax.load( blobUrlSrc, function( arrayBuffer ) {
+				var data = new Uint8Array( arrayBuffer ),
+					imageType = getImageTypeFromSignature( data.subarray( 0, 4 ) ),
```

Comment:
Shouldn't we pass here the whole `arrayBuffer`? It seems like taking the first 4 characters is already done in `getImageTypeFromSignature` function? In that case, this should be probably renamed again to just `getImageType`.